GeoLib/PointVec.h

```diff
@@ -143,6 +145,10 @@ class PointVec : public TemplateVec<Point>
 	 */
 	std::vector<std::size_t> _pnt_id_map;
 
+	/// The reverse map to the name to id map, for fast lookup of the name to a
+	/// given point id.
+	std::vector<std::string> _id_to_name_map;
```

Comment:
I additionally analysed Tom's test program in valgrind's massif tool for heap memory profiling.
 - For 1e5 points, all with names, we are speaking about 1MiB heap memory usage, which is not that important IMO.
 - The vector version has same memory usage as the master, because the maximum memory is used in the makePointsUnique function, which frees some memory afterwards. This additional memory is used by the `std::vector<std::string>`.
 - The map version indeed uses less memory up to 1e3 names. Minimum is for 0 names around 300KiB.

Summary: I tend to use the vector version because of simpler implementation and faster access compared to the map. The memory usage issues here are within reasonable limits.